substrate/frame/staking/src/slashing.rs

```diff
@@ -336,31 +332,17 @@ fn add_offending_validator<T: Config>(stash: &T::AccountId, disable: bool) {
 
 		let validator_index_u32 = validator_index as u32;
 
-		match offending.binary_search_by_key(&validator_index_u32, |(index, _)| *index) {
+		if let Err(index) = offending.binary_search_by_key(&validator_index_u32, |index| *index) {
 			// this is a new offending validator
-			Err(index) => {
```

Comment:
One more instance where we are refactoring away generality. We should be able to ensure in the code and through configs that this code path is never reached without removing it altogether (keeping in mind that we may want to implement a different disablement strategy in the future and we will use this code path).

Reply:
Okay but why should we pollute the code base with a dead code?
`ForceEra::<T>` is not removed. If someone wants to use it in the future - it's there. We don't know what we want to do in the future so better not keep dead code.